api/health/handler.go

```diff
@@ -45,14 +47,32 @@ func NewGetAndPostHandler(log logging.Logger, reporter Reporter) (http.Handler,
 	return handler, err
 }
 
+type errorMsg struct {
+	Error string `json:"error"`
+}
+
 // NewGetHandler return a health handler that supports GET requests reporting
 // the result of the provided [reporter].
-func NewGetHandler(reporter func() (map[string]Result, bool)) http.Handler {
+func NewGetHandler(reporter func(tags ...string) (map[string]Result, bool)) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// Make sure the content type is set before writing the header.
 		w.Header().Set("Content-Type", "application/json")
 
-		checks, healthy := reporter()
+		subnetIDs := r.URL.Query()["subnetID"]
+
+		// check if the subnetID is a valid ID
+		for _, subnetID := range subnetIDs {
+			if _, err := ids.FromString(subnetID); err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				err := fmt.Errorf("invalid subnetID %s: %w", subnetID, err)
+				_ = stdjson.NewEncoder(w).Encode(errorMsg{
+					Error: err.Error(),
+				})
+				return
+			}
+		}
```

Comment:
I think we can just remove this if we remove the requirement for the APIs to expect IDs

Reply:
yup, removed